ReSharper.FSharp/src/FSharp.Psi/src/Annotations/FSharpAnnotatedCacheProcessor.cs

```diff
@@ -0,0 +1,138 @@
+using System.Collections.Generic;
+using JetBrains.Annotations;
+using JetBrains.Metadata.Reader.API;
+using JetBrains.Metadata.Utils;
+using JetBrains.ReSharper.Plugins.FSharp.Psi.Impl;
+using JetBrains.ReSharper.Plugins.FSharp.Psi.Tree;
+using JetBrains.ReSharper.Psi;
+using JetBrains.ReSharper.Psi.Caches.AnnotatedEntities;
+using JetBrains.ReSharper.Psi.ExtensionsAPI;
+using JetBrains.ReSharper.Psi.Tree;
+
+namespace JetBrains.ReSharper.Plugins.FSharp.Psi.Annotations
+{
+  [Language(typeof(FSharpLanguage))]
+  public class FSharpAnnotatedMembersCacheProcessor : IAnnotatedEntitiesCacheProcessor
+  {
+    public void Process(IFile file, HashSet<string> attributeNames, AnnotatedEntitiesSet context)
+    {
+      var fsFile = file as IFSharpFile;
+      fsFile?.ProcessThisAndDescendants(new Processor(attributeNames, context));
+    }
+
+    private class Processor : TreeNodeVisitor, IRecursiveElementProcessor
+    {
+      private readonly HashSet<string> myAttributeNames;
+      private readonly AnnotatedEntitiesSet myContext;
+
+      public Processor(HashSet<string> attributeNames, AnnotatedEntitiesSet context)
+      {
+        myAttributeNames = attributeNames;
+        myContext = context;
+      }
+
+      public bool ProcessingIsFinished => false;
+      public bool InteriorShouldBeProcessed(ITreeNode element) => element is not IChameleonNode;
```

Comment:
We should probably also cover cases where parameters are inside a chameleon.
```fsharp
let f x =
    fun ([<A>] y) ->
        ()
```
and
```fsharp
let f =
    fun ([<A>] y) ->
        ()
```

That would need an addition to `ChameleonExpressionUtil` too.

Reply:
We also need to support probably the same case for ILocalBinding. And a simpler case with the regular local binding function.